src/weather_gen.h

```diff
@@ -5,10 +5,31 @@
 #include <string>
 
 #include "calendar.h"
+#include "enums.h"
 
 struct tripoint;
 class JsonObject;
 
+using weather_cache_key_type = std::pair<tripoint, time_point>;
+
+namespace std
+{
+template <>
+struct hash<weather_cache_key_type> {
+    std::size_t operator()( const weather_cache_key_type &k ) const {
+        constexpr uint64_t a = 2862933555777941757;
+        size_t result = k.first.z;
```

Comment:
You could start by setting `result` to `hash<tripoint>()(k.first)` rather than repeating that code here.

Reply:
Thanks, I didn't pay attention there ids hash for tripoints already.